Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.cpp

```diff
@@ -16,15 +16,17 @@ namespace AzToolsFramework
         AZStd::string name,
         AZStd::string description,
         AZStd::string category,
+        AZStd::string iconPath,
         AZStd::function<void()> handler,
-        AZStd::function<bool()> updateCallback
-        )
+        AZStd::function<bool()> checkStateCallback)
```

Comment:
nice, didn't know about this optimization for temporaries.

Reply:
Due to how the API can be called by different tools in different contexts (for example via Python), we enforce the passed function/lambda to be captured to gain ownership.